CPAC/pipeline/random_state/seed.py

```diff
@@ -0,0 +1,188 @@
+from nipype.interfaces.fsl.maths import MathsCommand
+from nipype.interfaces.fsl.utils import ImageMaths
+
+from CPAC.registration.utils import hardcoded_reg
+from CPAC.utils.interfaces.ants import AI
+from CPAC.utils.monitoring.custom_logging import set_up_logger
+
+_seed = {'seed': None}
+
+
+def random_random_seed():
+    '''Returns a random postive integer up to 2147483647
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    random_seed : int
+
+    Examples
+    --------
+    >>> 0 < random_random_seed() <= np.iinfo(np.int32).max
+    True
+    '''
+    return random.randint(1, np.iinfo(np.int32).max)
+
+
+def random_seed():
+    '''Function to access current random seed
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    seed : int or None
+    '''
+    if _seed['seed'] == 'random':
+        _seed['seed'] = random_random_seed()
+    return _seed['seed']
+
+
+def random_seed_flags():
+    '''Function to return dictionary of flags with current random seed.
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    dict of {'functions': {function: function}, 'interfaces': {list or tuple}}
+        Developer note: sequence matters here! Only the first match
+        will be applied!
+
+        In the 'functions' sub-dictionary, each key should be a function
+        used in a :py:class:`CPAC.utils.interfaces.function.Function`,
+        and each value should be a function that takes a single
+        parameter (a function) to apply to the key-function such that,
+        once the value-function is applied, the random seed is applied
+        when the key-function is run.
+
+        In the 'interfaces' sub-dictionary, each key should be an
+        :py:class:`nipype.interfaces.base.core.Interface`, and each
+        value should be either a list of strings to add to the
+        interface's flags/args or a tuple of (list of strings to add to,
+        list of strings to remove from) the interface's flags/args.
+
+    Examples
+    --------
+    >>> list(random_seed_flags().keys())
+    ['functions', 'interfaces']
+    >>> all([isinstance(random_seed_flags()[key], dict) for key in [
+    ...     'functions', 'interfaces']])
+    True
+    >>> set_up_random_state('random')
+    >>> list(random_seed_flags().keys())
+    ['functions', 'interfaces']
+    >>> all([isinstance(random_seed_flags()[key], dict) for key in [
+    ...     'functions', 'interfaces']])
+    True
+    '''
+    seed = random_seed()
+    if seed is None:
+        return {'functions': {}, 'interfaces': {}}
+    return {
+        'functions': {
+            # function: lambda function to apply to function source
+            hardcoded_reg: lambda fn_string: fn_string.replace(
+                'regcmd = ["antsRegistration"]',
+                f'regcmd = ["antsRegistration", "--random-seed", \"{seed}\"]'
+            )
+        },
+        'interfaces': {
```

Comment:
Perhaps add a note in the dev docs somewhere that this is where we need to add new interfaces/modules for which we want to control the random state?

Reply:
https://github.com/FCP-INDI/fcp-indi.github.io/pull/271